src/components/VmDetails/cards/SnapshotsCard/SnapshotDetail.js

```diff
@@ -0,0 +1,154 @@
+
+  const disksToRender = sortDisksForDisplay(snapshot.get('disks', Immutable.fromJS([])))
+  const showMoreDisks = disksToRender.size > 2
+  const diskToShow = disksToRender.slice(0, 2)
+  const additionalDisk = disksToRender.slice(2)
+
+  return <Popover
+    id={`snapshot-popover-${snapshot.get('id')}`}
+    title={snapshot.get('description')}
+    bsClass={`${style['popover']} popover`}
+    {...otherProps}
+  >
+    <div className={style['snapshot-detail-container']}>
+      <dl className={style['snapshot-properties']}>
+        <dt>
+          {msg.created()}
+        </dt>
+        <dd>
+          {`${time.time} ${time.date}`}
+        </dd>
+        <dt>
+          {msg.operatingSystem()}
+        </dt>
+        <dd>
+          {getOsHumanName(snapshot.getIn(['vm', 'os', 'type']))}
+        </dd>
+        <dt>
+          {msg.memory()}
+        </dt>
+        <dd>
+          {userFormatOfBytes(snapshot.getIn(['vm', 'memory', 'total'])).str} {snapshotMemoryState}
+        </dd>
+        <dt>
+          {msg.cpus()}
+        </dt>
+        <dd>
+          {snapshot.getIn(['vm', 'cpu', 'vCPUs'])}
+        </dd>
+        <dt>
+          {msg.status()}
+        </dt>
+        <dd>
+          {statusMap[snapshot.get('status')]}
+        </dd>
+      </dl>
+      <dl className={style['snapshot-properties']}>
+        <dt>
+          {msg.template()}
+        </dt>
+        <dd>
+          {template ? templateNameRenderer(template) : ''}
+        </dd>
+        <dt>
+          {msg.cdromBoot()}
+        </dt>
+        <dd>
+          {snapshot.getIn(['vm', 'cdrom', 'file', 'id']) ? snapshot.getIn(['cdrom', 'file', 'id']) : msg.empty() }
+        </dd>
+        <dt>
+          {msg.cloudInit()}
+        </dt>
+        <dd>
+          {getStatus(snapshot.getIn(['vm', 'cloudInit', 'enabled']))}
+        </dd>
+        <dt>
+          {msg.bootMenu()}
+        </dt>
+        <dd>
+          {getStatus(snapshot.getIn(['vm', 'bootMenuEnabled']))}
+        </dd>
+        <dt>
+          {msg.disks()}
+        </dt>
+        <dd>
+          <div className={style['snapshot-disk-list']}>
+            {diskToShow && diskToShow.map(diskRender)}
+            {
+              showMoreDisks &&
+              <OverlayTrigger
+                overlay={<Tooltip id={`snapshot-disk-tooltip-${snapshot.get('id')}`}>
+                  {
+                    additionalDisk && additionalDisk.map(diskRender)
+                  }
+                </Tooltip>}
+                placement='bottom'
+                trigger='click'
+                rootClose
+                key='info'
+              >
+                <a>Show more</a>
+              </OverlayTrigger>
+            }
+          </div>
+        </dd>
+      </dl>
+    </div>
+    <div style={{ textAlign: 'left' }}><RestoreConfirmationModal snapshot={snapshot} vmId={vmId}><Button bsStyle='default'>{ msg.restoreSnapshot() }</Button></RestoreConfirmationModal></div>
```

Comment:
nit-pick: Can you wrap this a bit to cut down on horizontal scrolling when reading the code?  Thanks.